portfolio/src/main/webapp/script.js

```diff
@@ -27,3 +27,13 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+function getHelloMessage(){
+
+    fetch("/data").then(response => response.text()).then((HelloMessage) => {
```

Comment:
I do like the code here and how you chained this together – it's simple and elegant. I do think a reader would still benefit from having a brief comment to say what's going on, either above this line, or above the getHelloMessage line.

Reply:
Thanks. I will add the comment!